hamiltonian_flow/models/harmonic_oscillator.py

```diff
@@ -0,0 +1,249 @@
+from functools import cached_property
+from typing import Dict, Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+from pydantic import BaseModel, computed_field
+
+
+class HarmonicOscillatorSystem(BaseModel):
+    """The params for the harmonic oscillator
+
+    :param omega: angular frequency of the harmonic oscillator
+    :param amplitude: the amplitude of the oscillation
+    :param phi: initial phase
+    """
+
+    omega: float
+    zeta: float = 0.0
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def period(self) -> float:
+        """period of the oscillator"""
+        return 2 * np.pi / self.omega
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def frequency(self) -> float:
+        """frequency of the oscillator"""
+        return 1 / self.period
+
+    @computed_field  # type: ignore[misc]
+    @cached_property
+    def type(
+        self,
+    ) -> Literal["simple", "under_damped", "critical_damped", "over_damped"]:
+        """which type of harmonic oscillators"""
+        if self.zeta == 0:
```

Comment:
case `< 0`

Reply:
Ah right. Wanted to write a validator in `HarmonicOscillatorSystem` to guarantee `zeta >= 0`

Reply:
Fixed in 794a493